### ISSUE TITLE
Waze crashed under SailfishOS alien-dalvik

### ISSUE THREAD

**Issue by vermut** [microg/android_packages_apps_GmsCore#107]
```
I/WAZE.STOPWATCH APP START(18844): $ Render # 3 $. Delta: 3440
I/WAZE    (18844): getConfig - event
I/WAZE    (18844): getConfig - callback
I/WAZE    (18844): getConfig - event
I/WAZE    (18844): getConfig - event
W/ActivityManager(32620): Permission Denial: Accessing service ComponentInfo{com.google.android.gms/org.microg.gms.gcm.PushRegisterService} from pid=18844, uid=10011 requires com.google.android.c2dm.permission.RECEIVE
D/AndroidRuntime(18844): Shutting down VM
W/dalvikvm(18844): threadid=1: thread exiting with uncaught exception (group=0x413d7360)
E/WAZE    (18844): com.waze.MainActivity@41e92cb0
E/AndroidRuntime(18844): FATAL EXCEPTION: main
E/AndroidRuntime(18844): java.lang.SecurityException: Not allowed to start service Intent { act=com.google.android.c2dm.intent.REGISTER (has extras) } without permission com.google.android.c2dm.permission.RECEIVE
E/AndroidRuntime(18844): 	at android.app.ContextImpl.startService(ContextImpl.java:1167)
E/AndroidRuntime(18844): 	at android.content.ContextWrapper.startService(ContextWrapper.java:383)
E/AndroidRuntime(18844): 	at com.waze.push.Registrar.register(Registrar.java:101)
E/AndroidRuntime(18844): 	at com.waze.push.Registrar.updateConfigItems(Registrar.java:77)
E/AndroidRuntime(18844): 	at com.waze.ConfigManager$1.callback(ConfigManager.java:78)
E/AndroidRuntime(18844): 	at com.waze.ifs.async.RunnableCallback$3.run(RunnableCallback.java:76)
E/AndroidRuntime(18844): 	at android.os.Handler.handleCallback(Handler.java:615)
E/AndroidRuntime(18844): 	at android.os.Handler.dispatchMessage(Handler.java:92)
E/AndroidRuntime(18844): 	at android.os.Looper.loop(Looper.java:137)
E/AndroidRuntime(18844): 	at android.app.ActivityThread.main(ActivityThread.java:4781)
E/AndroidRuntime(18844): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime(18844): 	at java.lang.reflect.Method.invoke(Method.java:511)
E/AndroidRuntime(18844): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:805)
E/AndroidRuntime(18844): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:568)
E/AndroidRuntime(18844): 	at dalvik.system.NativeStart.main(Native Method)
W/ActivityManager(32620):   Force finishing activity com.waze/.MainActivity
I/WAZE.STOPWATCH APP START(18844): $ Render # 4 $. Delta: 3717
I/WAZE.STOPWATCH APP START(18844): $ Render # 5 $. Delta: 3926
W/ActivityManager(32620): Activity pause timeout for ActivityRecord{41ffe8b0 surfaceId:4 com.waze/.MainActivity}

```

Worked before in gapps-free times. Is it forcing me to run GSF?

**Comment by vermut** [comment]
After installing GSF and reinstalling to latest version 4.0.0.2 it became
```
/GmsHttpFormClient(20407): -- Request --
D/GmsHttpFormClient(20407): app=com.waze&cert=35b438fe1bc69d975dc8702dc16ab69ebf65f26f&sender=1005457359603&X-GOOG.USER_AID=0&device=0&app_ver=1020913
D/dalvikvm(21053): GC_FOR_ALLOC freed 1827K, 17% free 13685K/16455K, paused 27ms, total 27ms
D/dalvikvm(21053): GC_CONCURRENT freed 1737K, 17% free 13802K/16455K, paused 2ms+7ms, total 56ms
D/dalvikvm(21053): WAIT_FOR_CONCURRENT_GC blocked 42ms
D/GmsHttpFormClient(20407): -- Response --
D/GmsHttpFormClient(20407): Error=PHONE_REGISTRATION_ERROR
D/GmsGcmRegisterSvc(20407): received response: RegisterResponse{token='null', retryAfter='null', deleted='null'}
D/GmsGcmRegisterSvc(20407): register[res]: Intent { act=com.google.android.c2dm.intent.REGISTRATION (has extras) } extras=Bundle[{error=SERVICE_NOT_AVAILABLE}]
E/WAZE    (21053): RegistrationIntentService: GCM: An exception occurred in RegistrationIntentService while trying to handle intent SERVICE_NOT_AVAILABLE java.io.IOException: SERVICE_NOT_AVAILABLE
E/WAZE    (21053): 	at com.google.android.gms.iid.Rpc.zzb(Unknown Source)
E/WAZE    (21053): 	at com.google.android.gms.iid.Rpc.zza(Unknown Source)
E/WAZE    (21053): 	at com.google.android.gms.iid.InstanceID.zzb(Unknown Source)
E/WAZE    (21053): 	at com.google.android.gms.iid.InstanceID.getToken(Unknown Source)
E/WAZE    (21053): 	at com.waze.push.RegistrationIntentService.onHandleIntent(RegistrationIntentService.java:43)
E/WAZE    (21053): 	at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
E/WAZE    (21053): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/WAZE    (21053): 	at android.os.Looper.loop(Looper.java:137)
E/WAZE    (21053): 	at android.os.HandlerThread.run(HandlerThread.java:60)

```

**Comment by vermut** [comment]
Finally worked after disabling GCM in microG settings.

**Comment by mar-v-in** [comment]
GSF should not be required, but reinstalling Waze should have solved the first issue in your case. The second issue appears as your device was not checked in to Google services yet. This can happen when checkin is disabled or was just enabled. You can force a checkin by dialing `*#*#CHECKIN#*#*` on the system dialer.

**Comment by mar-v-in** [comment]
This should be fixed as of 1614289 / v0.2.2
